frontend/src/metabase/common/components/FilterPicker/NumberFilterPicker/constants.ts

```diff
@@ -0,0 +1,13 @@
+import type { NumberFilterValueCountMap } from "./types";
+
+export const numberFilterValueCountMap: NumberFilterValueCountMap = {
+  "is-null": 0,
+  "not-null": 0,
+  ">": 1,
+  ">=": 1,
+  "<": 1,
+  "<=": 1,
+  between: 2,
+  "=": "multiple",
+  "!=": "multiple",
+};
```

Comment:
This is essentially the same as, but way more performant than, mapping through `filterableColumnOperators` and calling `displayInfo` on each one. it also provides us *more* type safety because it ensures that we've handled the exact set of NumberFilterOperatorNames, where `displayInfo` will only give us FilterOperatorName as a type (which includes invalid filters for numbers).

Also, we never set an operator based on this, the actual operatorName always comes from `filterableColumnOperators`, this just tells us how many inputs the selected operator gets.